### PR TITLE
Pz/types 007 - EarningRules

### PR DESCRIPTION
**Pull request by p-zielinski**
---
'@voucherify/sdk': minor
---

Add support for 3 missing endpoints in loyalties api. Corrected a typo in one of loyalties api method.
- Added support for new endpoints: `GET /loyalties/{campaignId}/earning-rules/{earningRuleId}`, `POST /loyalties/{campaignId}/earning-rules/{earningRuleId}/enable` and `POST /loyalties/{campaignId}/earning-rules/{earningRuleId}/disable` [(examples available in readme.md)](..%2F..%2Fpackages%2Fsdk%2FREADME.md)
- New exported types/interfaces: `LoyaltiesGetEarningRuleResponseBody`, `LoyaltiesEnableEarningRulesResponseBody`, `LoyaltiesDisableEarningRulesResponseBody`, `EarningRuleEvent`, `EarningRuleBase`, `EarningRuleLoyaltyFixed`, `EarningRuleLoyaltyProportionalOrder`, `EarningRuleLoyaltyProportionalOrderAmount`, `EarningRuleLoyaltyProportionalOrderTotalAmount`, `EarningRuleLoyaltyProportionalOrderMetadata`, `EarningRuleLoyaltyProportionalOrderItems`, `EarningRuleLoyaltyProportionalOrderItemsQuantity`, `EarningRuleLoyaltyProportionalOrderItemsAmount`, `EarningRuleLoyaltyProportionalOrderItemsSubtotalAmount`, `EarningRuleLoyaltyProportionalCustomer`, `EarningRuleLoyaltyProportionalCustomEvent`, `EarningRuleLoyaltyProportional`

### CODE REVIEW

File: packages/sdk/src/types/Loyalties.ts [hunk 1]
```diff
@@ -497,3 +497,166 @@ export interface LoyaltyPointsTransfer {
...
+	| 'order.paid'
+	| 'customer.segment.entered'
+	| 'custom_event'
+	| 'customer.loyalty.tier.upgraded'
+	| 'customer.loyalty.tier.downgraded'
+	| 'customer.loyalty.tier.prolonged'
+	| 'customer.loyalty.tier.joined'
+	| 'customer.loyalty.tier.left'
+
+export interface EarningRuleLoyaltyFixed {
```
Comment (marcin-slezak) [inline packages/sdk/src/types/Loyalties.ts:561]: I just wonder about the naming - do we need the word `Loyalty` in `EarningRuleLoyaltyFixed` interface name? The same with all names below...

**Review by marcin-slezak — CHANGES_REQUESTED** [review]

File: packages/sdk/src/types/Loyalties.ts
[the remark this answers is not in the document]
Comment (p-zielinski) [reply packages/sdk/src/types/Loyalties.ts:1]: will be added in 1h

**Comment by p-zielinski** [comment]
https://docs.voucherify.io/v2018-08-01-piotr-482/reference/get-earning-rule

File: packages/sdk/src/types/Loyalties.ts
[the remark this answers is not in the document]
Comment (p-zielinski) [reply packages/sdk/src/types/Loyalties.ts:1]: https://docs.voucherify.io/v2018-08-01-piotr-482/reference/get-earning-rule

**Review by marcin-slezak — APPROVED** [review]

**Pull request merged**